project/PACKAGERS.md

```diff
@@ -35,6 +47,10 @@ If you would rather (or must, due to distro policy) package these dependencies
 yourself, take a look at "vendor.mod" for an easy-to-parse list of the
 exact version for each.
 
+Note that this project is **not** a Go module. To properly build the project,
+`GOPATH` mode must be used. The project is migrating to Go modules, but `vendor.mod`
```

Comment:
This seems like strange wording.
The issue with go.mod is more about people importing the project, not building it.
We absolutely use `vendor.mod` as a stand-in for `go.mod`:

https://github.com/moby/moby/blob/6fd893bd3af4f085d1bdbb059fc6a98ff2629967/hack/make/.binary#L85

We also don't use GOPATH mode (or rather don't require it), hence the `hack/with-go-mod.sh`.

Reply:
There are issues with go.mod in building; while we do some trickery with `vendor.mod` here, it's worth noting that renaming `vendor.mod` to `go.mod` doesn't work; this is a special case where the Go toolchain requires a stub `go.mod` to be bent the way we want.

In general I see a lot of https://github.com/moby/moby/issues/44618 coming up; when it comes to advice for packagers, I think we really should express GOPATH is the supported way to build.

Reply:
Our own build scripts are turning on module mode, though.
If someone builds with `hack/make.sh [binary|dynbyinary]` they are building with module mode.

Reply:
It can be used correctly with the wrapper for sure; I just don't know that is something that we want to expose. Do you think it's worth documenting how to build in module mode (with the wrapper/correct arguments) and making that a "stable" interface for people to depend on?

Reply:
`hack/make.sh` is the interface we have always expected people to use.
I think it is fair to mention that in this doc?